src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.models.BlobItem;
+import org.slf4j.Logger;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@SuppressWarnings("java:S1135") // ignore TODOs. will be removed when blob processing part is implemented
+public class ContainerProcessor {
+
+    private static final Logger LOGGER = getLogger(ContainerProcessor.class);
+
+    public void process(BlobContainerAsyncClient containerClient) {
+        String containerName = containerClient.getBlobContainerName();
+        AtomicInteger processedBlobCount = new AtomicInteger(0);
+
+        LOGGER.info("Processing container {}", containerName);
+
+        containerClient
+            .listBlobs()
+            .subscribe(
+                blob -> {
+                    processBlob(blob);
+                    processedBlobCount.incrementAndGet();
```

Comment:
Blob may not be processed due to not job not being able to acquire a lock, but the counter will be increased here anyway. Not what we want I assume.

Reply:
updated the log message. `processedBlobCount` variable is removed